src/stores/zapPendingOperations/ZapPendingOperationsStore.ts

```diff
@@ -0,0 +1,199 @@
+import { shallow } from 'zustand/shallow';
+import { persist, createJSONStorage } from 'zustand/middleware';
+import {
+  WalletMethods,
+  WalletMethodArgsType,
+  WalletMethodReturnType,
+} from 'src/providers/ZapInitProvider/types';
+import { createWithEqualityFn } from 'zustand/traditional';
+
+interface PendingOperationData<T extends WalletMethods> {
+  operationName: T;
+  args: WalletMethodArgsType<T>;
+  timestamp: number;
+  id: string;
+}
+
+interface PromiseResolver<T extends WalletMethods> {
+  resolve: (value: WalletMethodReturnType<T>) => void;
+  reject: (error: Error) => void;
+}
+
+const serializeWithTypes = (obj: any): any => {
+  if (typeof obj !== 'object' || obj === null) {
+    return {
+      value: obj,
+      type: typeof obj,
+    };
+  }
+
+  if (Array.isArray(obj)) {
+    return {
+      value: obj.map(serializeWithTypes),
+      type: 'array',
+    };
+  }
+
+  const result: any = {};
+
+  for (const [key, value] of Object.entries(obj)) {
+    if (typeof value === 'bigint') {
+      result[key] = {
+        value: value.toString(),
+        type: 'bigint',
+      };
+    } else if (typeof value === 'object' && value !== null) {
+      result[key] = serializeWithTypes(value);
+    } else {
+      result[key] = {
+        value: value,
+        type: typeof value,
+      };
+    }
+  }
+
+  return result;
+};
+
+const deserializeWithTypes = <T>(obj: any): T => {
+  // Check if this is a typed value object
+  if (
+    typeof obj === 'object' &&
+    obj !== null &&
+    'value' in obj &&
+    'type' in obj
+  ) {
+    const typedValue = obj as { value: any; type: string };
+
+    if (typedValue.type === 'bigint') {
+      return BigInt(typedValue.value) as T;
+    } else if (typedValue.type === 'array') {
+      return typedValue.value.map(deserializeWithTypes) as T;
+    } else {
+      // For primitive types, just return the value
+      return typedValue.value as T;
+    }
+  }
+
+  // If it's not a typed object, it might be a plain object (shouldn't happen with new structure)
+  if (typeof obj === 'object' && obj !== null) {
+    const result: Record<string, unknown> = {};
+
+    for (const [key, value] of Object.entries(obj)) {
+      result[key] = deserializeWithTypes(value);
+    }
+
+    return result as T;
+  }
+
+  return obj as T;
+};
```

Comment:
Serialization / deserialization is alway a bit tricky to get right (dealing with collisions, weird inputs, etc), and it's always too late when you find out about a bug.

Since they are pure functions, could we move them to their module and unit-test them a bit? Or would it make sense to use a library like [superjson](https://github.com/flightcontrolhq/superjson)?

Reply:
That is an amazing point @laurentsenta let's go with `superjson` ❤️ though would love to do it in a separate PR